app/controlplane/internal/service/auth.go

```diff
@@ -304,7 +306,10 @@ func callbackHandler(svc *AuthService, w http.ResponseWriter, r *http.Request) *
 	}
 
 	// Create user if needed
-	u, err := svc.userUseCase.FindOrCreateByEmail(ctx, claims.preferredEmail())
+	u, err := svc.userUseCase.FindOrCreateByEmail(ctx, claims.preferredEmail(), &biz.FindOrCreateByEmailOpts{
+		FirstName: &claims.GivenName,
```

Comment:
Don't assume these claims will have values always. It's up to the user to give consent.

Reply:
Yes, that's why those fields are pointers and in the case they are set empty, there is a check in the biz layer that prevents the update of those fields. I'll ping you there.